jaeger-core/src/main/java/io/jaegertracing/Configuration.java

```diff
@@ -137,6 +136,11 @@
    */
   public static final String JAEGER_PROPAGATION = JAEGER_PREFIX + "PROPAGATION";
 
+  /**
+   * The fully qualified class name for the sender factory
```

Comment:
Just wondering if we could select the factory based on a simpler name, which may be maps onto a predefined class name - so the configuration by the user requires less implementation knowledge?

Reply:
This would be required only if there are multiple factories in the classpath. Do you think it would be a common use case to justify a map of "common" factories?

Reply:
Not necessary common, but just thinking of the detail the user would need to know (i.e. the fully qualified class name) to select a transport.

Reply:
In the usual case, they just need to have the transport on the classpath, there's no need to know the class name. If they end up somehow with two transports, the class name is used to disambiguate. 

How about we go without this map for now, and add later if the case is more common than we thought?

Reply:
Two other suggestions - (1) we treat multiple factories as an error - so if the case comes up to justify having two in the classpath, we can explore it further then, (2) add a type property on the factory, and that is what is specified in the environment variable?

Reply:
> add a type property on the factory, and that is what is specified in the environment variable?

I like this idea, as it's easily extensible.

Reply:
> I think FQN is better than just a class name.

This is also available: if the env var contains a class name, we try to load it. Otherwise, we try to get a single factory from the service loader. If there are multiple, we try to match the type with the env var. If none matches, NoopSender is returned.